homeassistant/components/transmission/sensor.py

```diff
@@ -54,15 +106,23 @@ async def async_setup_entry(
 class TransmissionSensor(SensorEntity):
     """A base class for all Transmission sensors."""
 
+    _attr_has_entity_name = True
     _attr_should_poll = False
 
-    def __init__(self, tm_client, client_name, sensor_name, sub_type=None):
+    def __init__(
+        self,
+        entity_description: SensorEntityDescription,
+        tm_client: TransmissionClient,
+        client_name: str,
+    ) -> None:
         """Initialize the sensor."""
+        self.entity_description = entity_description
         self._tm_client: TransmissionClient = tm_client
-        self._client_name = client_name
-        self._name = sensor_name
-        self._sub_type = sub_type
-        self._state = None
+        self._state: Any | None = None
+
+        self._attr_unique_id = (
+            f"{self._tm_client.api.host}-{client_name} {self.entity_description.key}"
+        )
```

Comment:
This isn't allowed for a unique ID.

See: <https://developers.home-assistant.io/docs/entity_registry_index#unacceptable-sources-for-a-unique-id>

Reply:
Sure, but this violates our design rules. I cannot accept or approve this.

Reply:
Does it mean that no PRs affecting Transmission in any way can be merged until unique id migration is done?